### PR TITLE
Upgrade to cosmos-sdk v0.40.0-rc6

### PR DESCRIPTION
**Pull request by alpe**
Upgrade to [`cosmos-sdk v0.40.0-rc6`](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.40.0-rc6)

There is a known 🐛 [#8251](https://github.com/cosmos/cosmos-sdk/issues/8251) with this SDK version in the gentx CLI command. The workaround is documented in the issue as well:
```sh
# pass the staked amount as argument AND as parameter
wasmd gentx validator "250000000$STAKE" --chain-id="$CHAIN_ID" --amount="250000000$STAKE"
```

### CODE REVIEW

File: contrib/local/setup_wasmd.sh [hunk 1]
```diff
@@ -22,5 +22,5 @@ for addr in "$@"; do
   wasmd add-genesis-account "$addr" "1000000000$STAKE,1000000000$FEE"
 done
 # submit a genesis validator tx
-(echo "$PASSWORD"; echo "$PASSWORD"; echo "$PASSWORD") | wasmd gentx validator --chain-id="$CHAIN_ID" --amount "250000000$STAKE"
+(echo "$PASSWORD"; echo "$PASSWORD"; echo "$PASSWORD") | wasmd gentx validator "250000000$STAKE" --chain-id="$CHAIN_ID"
```
Comment (ethanfrey) [inline contrib/local/setup_wasmd.sh:25]: Oh, this is important part for the cosmjs tests.
I don't know how best to flag it, so @willclarktech @webmaster128 Note this when we test cosmjs against 0.14.0 wasmd

File: go.mod [hunk 2]
```diff
@@ -4,8 +4,8 @@ go 1.15
 
 require (
 	github.com/CosmWasm/wasmvm v0.12.0
-	github.com/cosmos/cosmos-sdk v0.40.0-rc5
-	github.com/cosmos/iavl v0.15.0-rc5
+	github.com/cosmos/cosmos-sdk v0.40.0-rc6
+	github.com/cosmos/iavl v0.15.3
```
Comment (ethanfrey) [inline go.mod:8]: Ohh... one less rc dependency :smile: 

Faster than dependabot :wink:

File: x/wasm/internal/keeper/staking_test.go [hunk 3]
```diff
@@ -616,7 +616,7 @@ func addValidator(t *testing.T, ctx sdk.Context, stakingKeeper stakingkeeper.Kee
 	pubKey := privKey.PubKey()
 	addr := sdk.ValAddress(pubKey.Address())
 
-	pkAny, err := codectypes.PackAny(pubKey)
+	pkAny, err := codectypes.NewAnyWithValue(pubKey)
```
Comment (ethanfrey) [inline x/wasm/internal/keeper/staking_test.go:619]: This is the only needed code change. Cool.

**Review by ethanfrey — APPROVED** [review]
Good stuff.
Not sure how to document the client changes (diff staking args).

Maybe a little changelog entry?

File: contrib/local/setup_wasmd.sh (hunk 1 above)
Comment (webmaster128) [reply contrib/local/setup_wasmd.sh:25]: Does this change anything for the caller of this script? We should get the change when bundled as part of the docker container.

Comment (alpe) [reply contrib/local/setup_wasmd.sh]: yes, this is part of the docker script as well.
Please note: due to the 🐛  in the sdk the workaround is to pass the staked tokens twice. As argument and as flag.

**Comment by alpe** [comment]
I had added https://github.com/CosmWasm/docs/issues/128 to update the docs

**Pull request merged**